### DISCUSSION TITLE
Guide to Building

### DISCUSSION THREAD

**Discussion by NWoodsman** [dotnet/wpf#9079] [General]
I can't find a guide to clarify steps needed to build from source. My understanding is that it requires building from source and providing additional dll's of MS internal libraries that haven't been opened.

Please provide some direction, thank you.

**Comment by miloush** [comment]
It really is a matter of cloning the repo and calling `build` from a developer command prompt (I think at some point it had to run as admin for me.) If that doesn't work, feel free to search and/or create new issues. There is https://github.com/dotnet/wpf/wiki/Building-WPF-Locally and also couple of useful documents under https://github.com/dotnet/wpf/tree/main/Documentation.
